sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/pipeline/HttpLoggingPolicy.java

```diff
@@ -294,7 +292,7 @@ private static String getRedactedUrl(URL url, Set<String> allowedQueryParameterN
     private static void addHeadersToLogMessage(List<HttpHeaderName> allowedHeaderNames, HttpHeaders headers,
                                                ClientLogger.LoggingEventBuilder logBuilder) {
         for (HttpHeader header : headers) {
-            String headerName = header.getName();
+            String headerName = header.getName().toString();
```

Comment:
We can use HttpHeaderName here instead of String. addKeyValue should stringify the value and the check for if the header can be logged uses HttpHeaderName (also we should look into changing the List to a Set)